articles/divide-and-conquer-algorithms/index.md

```diff
@@ -0,0 +1,177 @@
+    if(len(price_list)<2):
+        raise ValueError('Getting a profit requires at least 2 prices')
+    
+    maximum_profit = 0
+    if start >= end:
+        return maximum_profit
+    for buy in range(start, end):
+        for sell in range(buy+1, end+1):
+            tentative_profit = price_list[sell] - price_list[buy] + maximize_profit(price_list, start, buy - 1) + maximize_profit(price_list, sell, end)
+            maximum_profit = max(maximum_profit, tentative_profit)
+        
+    return maximum_profit
+
+# inputs
+price1 = [1, 5, 3, 2]
+price2 = [7, 2, 8, 9]
+price3 = [1, 6, 7, 9]
+price4 = [9, 7, 4, 1]
+price5 = [10, 5, 1, 0]
+price6 = [100, 180, 260, 310, 40, 535, 695]
+
+print("     List", "     --> Profit(₹)")
+print("1.", price1, "--> ₹", maximize_profit(price1,0,len(price1)-1))
+print("2.", price2, "--> ₹", maximize_profit(price2,0,len(price2)-1))
+print("3.", price3, "--> ₹", maximize_profit(price3,0,len(price3)-1))
+print("4.", price4, "--> ₹", maximize_profit(price4,0,len(price4)-1))
+print("5.", price5, "--> ₹", maximize_profit(price5,0,len(price5)-1))
+print("6.", price6, "--> ₹", maximize_profit(price6,0,len(price6)-1))
+```
+
+### Output 
+
+The program outputs the following profits:
+
+```bash
+     List      --> Profit(₹)
+1. [1, 5, 3, 2] --> ₹ 4
+2. [7, 2, 8, 9] --> ₹ 7
+3. [1, 6, 7, 9] --> ₹ 8
+4. [9, 7, 4, 1] --> ₹ 0
+5. [10, 5, 1, 0] --> ₹ 0
+6. [100, 180, 260, 310, 40, 535, 695] --> ₹ 865
+```
+
+### Output Analysis
+
+Let us analyze the edge cases. This helps us in identifying the advantages and disadvantages of the divide and conquer approach.
+
+#### Case 1:
+
+*Input List*: [1,5,3,2]
+
+*Output*: **Maximum Profit**: 4:
+**Buy** at **1**:
+**Sell** at **5**
+
+*Best Profit Possible*: 4
+
+#### Case 2:
+*Input List*: [9,7,4,1]
+
+*Output*: **Maximum Profit**: 0 :**Buy** at **9**
+:**Sell** at **7**
+
+*Best Profit Possible*: -2
+
+In this case, the output remains at 0 and does not output negative numbers. Therefore, this approach can be used to find only the greatest _profit_ possible. 
+
+#### Case 3:
+
+*Input List:*
+[100,180,260,310,40,535,695]
+
+*Output*: **Maximum Profit**: 865
+
+*Optimal Solution*
+
+**Buy** at **100**, **Sell** at **310**
+**Buy** at **40**, Sell at **695**
+
+**Total Profit** = 865
+
+Compared to the greedy approach, this method gives the optimal solution to the problem. We observed that the maximum profit in greedy method was 655. In this case, the total profit is 865.
+
+### Advantages 
+- The program can work with many buy and sell operations due to its recurring property. Multiple buy and sell operations take place, and the best profits are returned 
+- Divide and Conquer algorithm's solutions are always optimal. The comparison of code output: scenario- 3 shows the same. The greedy algorithm outputs 655, whereas the divide and conquer algorithm outputs 865.
+
+### Disadvantages
+
+- The worst-case time complexity of the function `maximize_profit`()  is Θ(n^2*log(n)) 
```

Comment:
Could you please do a quick run through of how we arrive at the complexity? This might help readers who are new to algorithms (which includes complexity analysis)

Reply:
Well, I arrived at it computationally. This is a complex algorithm, and I would stay out of the derivation for now. I could not construct the recursive equations for this problem. I asked my profs at college, as well. It was a tough nut to crack. I will mention that I arrived at it computationally, will not go into the depths of deriving the same.